server.js

```diff
@@ -1,10 +1,106 @@
-const express = require('express'); // remember to install your npm packages
-
+const mongoose = require('mongoose');
+const express = require('express');
+const bodyParser = require('body-parser');
+mongoose.connect('mongodb://localhost/budgetDB');
+const Budget = require('./Models/budget.js');
+const Category = require('./Models/category.js');
+const Expense = require('./Models/expense.js');
 const server = express();
+const to = require('./to')
+
+server.use(bodyParser.json());
 
-// add your server code
+server.post('/budget', async (req, res) => {
+    const bud = req.body;
+    if (bud.title && bud.budgetAmount) {
+        const budget = new Budget(bud);
+        const [err] = await to(budget.save());
+        if (err) {
+            res.status(500).json({
+                error: 'budgetAmount or title is missing.'
+            })
+        } else {
+            res.status(200).json({
+                sucess: 'Budget got saved.'
+            });
+        }
+    } else {
+        res.status(422).json({
+            error: 'budgetAmount or title is missing.'
+        })
+    }
+});
+server.post('/category', (req, res) => {
+    const title = req.body;
+    if (title) {
+        const category = new Category(title);
```

Comment:
The way you've written this if condition, it will always evaluate to "true" - there will always be a value assigned to "title" (so long as a POST has been made to /categories). One way to check if an actual "title" key has been is to check the contents of the req.body object using something like the "hasOwnProperty" method, e.g.

```
> reqBody = { twitter: 'testCategory' }
{ twitter: 'testCategory' }
> reqBody.hasOwnProperty("Title")
false
```